### PR TITLE
Fix for issue #4254

### LINKED ISSUE #4254: Login form on CMS Toolbar should use `USERNAME_FIELD`
Django allows custom User models that user alternative fields instead of "username". Most common example of this is probably email-only User model, i.e. User model that does not have "username" field and instead relies on "email" field for identification.

To this end, Dajngo provides `USERNAME_FIELD`: https://docs.djangoproject.com/en/1.7/topics/auth/customizing/#django.contrib.auth.models.CustomUser.USERNAME_FIELD

Login form on the CMS Toolbar should follow this, and not hardcode "Username" for the form label. 
NB - the login form itself works fine, only the label is wrong.

`templates/cms/toolbar/items/login.html` seems like the place to start (or finish) this.

### PR DESCRIPTION
**Pull request by jproffitt**
Fix #4254

### CODE REVIEW

File: cms/toolbar/toolbar.py [hunk 1]
```diff
@@ -24,7 +23,7 @@
 
 
 class CMSToolbarLoginForm(AuthenticationForm):
-    username = forms.CharField(label=_("Username"), max_length=100)
+    username = forms.CharField(max_length=100)
```
Comment (yakky) [inline cms/toolbar/toolbar.py]: I assume we can completely remove this, it's already defined in ``AuthenticationForm``, right?

Comment (jproffitt) [inline cms/toolbar/toolbar.py]: Well yes, but the `username` in `AuthenticationForm` has a `max_length` of 254. I guess that should probably match whatever django sets. I wasn't sure if there was a reason it was set to 100 for the cms.

Comment (yakky) [inline cms/toolbar/toolbar.py]: Legacy code from the days django had very short usernames. django CMS does not define a user, so we can now get rid of it and just use whatever Django provides

Comment (jproffitt) [inline cms/toolbar/toolbar.py]: Ok, I have added a new commit to remove it.

**Comment by yakky** [comment]
@jproffitt awesome, thanks

**Pull request merged**